taiga_contrib_ldap_auth/services.py

```diff
@@ -62,17 +67,29 @@ def ldap_login_func(request):
     # (or any other attribute)
     login_input = request.DATA.get('username', None)
     pass_input = request.DATA.get('password', None)
-
-    # TODO: sanitize before passing to LDAP server?
-    username, email, full_name = connector.login(login = login_input,
-                                                 password = pass_input)
-
-    user = register_or_update(username = username,
-                              email = email,
-                              full_name = full_name)
-
-    data = make_auth_response_data(user)
-    return data
+    try:
+        # TODO: make sure these fields are sanitized before passing to LDAP server!
+        username, email, full_name = connector.login(login = login_input, password = password_input)
+    except connector.LDAPUserLoginError as ldap_error:
```

Comment:
I'm catching `LDAPUserLoginError` here because it specifically implies that the user/password combination is wrong. An `LDAPConnectionError` means that either the authenticated binding to the LDAP server failed (which is a configuration error) or the LDAP server failed some other way (which is also a configuration error). In both cases I don't think the code should try to authenticate against the local database, keeping in mind the Python Zen: *Errors should never pass silently*.

Does that make sense or did I miss something?